system/Validation/Rules.php

```diff
@@ -436,6 +436,47 @@ public function required_without(
         return true;
     }
 
+    /**
+     * This field is required when any of the other required fields have their expected values present
+     * in the data.
+     *
+     * Example (The special_option field is required when the normal_option,1,2 field has the given value.):
+     *
+     *     required_if[normal_option,1,2]
+     *
+     * @param string|null          $str
+     * @param string|null          $fieldWithValue that we should check if present
+     * @param array<string, mixed> $data           Complete list of field from the form
+     */
+    public function required_if($str = null, ?string $fieldWithValue = null, array $data = []): bool
+    {
+        if ($fieldWithValue === null || $data === []) {
+            throw new InvalidArgumentException('You must supply the parameters: field,expected_values, data.');
+        }
+
+        // Separate fields and expected values
+        $parts          = explode(',', $fieldWithValue);
+        $field          = array_shift($parts); // Get field
+        $expectedValues = $parts; // The remainder is the expected value
+
+        if (trim($field) === '' || $expectedValues === []) {
+            throw new InvalidArgumentException('You must supply the expected values of field: E.g. field,value1,value2,...');
```

Comment:
It may be better to have a more descriptive exception message by saying which field needs to be supplied with the expected values.

Reply:
how about this one?
```php
throw new InvalidArgumentException("You must supply the field value: {$field},value? or you can add multiple value like this {$field},value1,value2 and so on.");
```